xarray/tests/test_backends.py

```diff
@@ -1675,88 +1676,65 @@ class H5NetCDFDataTestAutocloseTrue(H5NetCDFDataTest):
     autoclose = True
 
 
-class OpenMFDatasetManyFilesTest(TestCase):
-    def validate_open_mfdataset_autoclose(self, engine, nfiles=10):
-        randdata = np.random.randn(nfiles)
-        original = Dataset({'foo': ('x', randdata)})
-        # test standard open_mfdataset approach with too many files
-        with create_tmp_files(nfiles) as tmpfiles:
-            for readengine in engine:
-                writeengine = (readengine if readengine != 'pynio'
-                               else 'netcdf4')
-                # split into multiple sets of temp files
-                for ii in original.x.values:
-                    subds = original.isel(x=slice(ii, ii + 1))
-                    subds.to_netcdf(tmpfiles[ii], engine=writeengine)
-
-                # check that calculation on opened datasets works properly
-                ds = open_mfdataset(tmpfiles, engine=readengine,
-                                    autoclose=True)
-                self.assertAllClose(ds.x.sum().values,
-                                    (nfiles * (nfiles - 1)) / 2)
-                self.assertAllClose(ds.foo.sum().values, np.sum(randdata))
-                self.assertAllClose(ds.sum().foo.values, np.sum(randdata))
-                ds.close()
-
-    def validate_open_mfdataset_large_num_files(self, engine):
-        self.validate_open_mfdataset_autoclose(engine, nfiles=2000)
+@pytest.fixture(params=[
+    pytest.mark.skipif(not has_scipy, 'scipy', reason='requires scipy'),
+    pytest.mark.skipif(not has_netCDF4, 'netcdf4', reason='requires netcdf4'),
+    pytest.mark.skipif(not has_h5netcdf, 'h5netcdf',
+                       reason='requires h5netcdf'),
+    pytest.mark.skipif(not has_pynio, 'pynio', reason='requires pynio')])
+def readengine(request):
+    return request.param
```

Comment:
Can anyone guess as to why the pynio tests are not being properly skipped using this fixture configuration? 

@mrocklin - I think you suggested this approach, which I like, but something weird is happening.

Reply:
I should note, this is working on python 2.7 and 3.4. Just not for 3.5 or 3.6.